dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+
+        esPathHome =
+                !new File(esPathHome).isAbsolute() ? FileUtil.getRealPath(esPathHome) : esPathHome;
+
+        return esPathHome;
+    }
+
+    public RestHighLevelClient getClient() {
+        return client;
+    }
+
+    public void setClient(final RestHighLevelClient theClient) {
+        if(theClient!=null) {
+            client = theClient;
+        }
+    }
+
+    public void close() throws IOException {
+        if(client!=null) {
+            client.close();
+        }
+    }
+
+    /*
+     * Copyright 2014 The Netty Project
+     *
+     * The Netty Project licenses this file to you under the Apache License,
+     * version 2.0 (the "License"); you may not use this file except in compliance
+     * with the License. You may obtain a copy of the License at:
+     *
+     *   http://www.apache.org/licenses/LICENSE-2.0
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+     * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+     * License for the specific language governing permissions and limitations
+     * under the License.
+     */
+    private static final class PemReader {
+
+        private static final Pattern CERT_PATTERN = Pattern.compile(
+                "-+BEGIN\\s+.*CERTIFICATE[^-]*-+(?:\\s|\\r|\\n)+" + // Header
+                        "([a-z0-9+/=\\r\\n]+)" +                    // Base64 text
+                        "-+END\\s+.*CERTIFICATE[^-]*-+",            // Footer
+                CASE_INSENSITIVE);
+
+        private static final Pattern KEY_PATTERN = Pattern.compile(
+                "-+BEGIN\\s+.*PRIVATE\\s+KEY[^-]*-+(?:\\s|\\r|\\n)+" + // Header
+                        "([a-z0-9+/=\\r\\n]+)" +                       // Base64 text
+                        "-+END\\s+.*PRIVATE\\s+KEY[^-]*-+",            // Footer
+                CASE_INSENSITIVE);
+
+        private PemReader() {
+
+        }
+
+        private static KeyStore loadTrustStore(File certificateChainFile)
+                throws IOException, GeneralSecurityException {
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null, null);
+
+            List<X509Certificate> certificateChain = readCertificateChain(certificateChainFile);
+            for (X509Certificate certificate : certificateChain) {
+                X500Principal principal = certificate.getSubjectX500Principal();
+                keyStore.setCertificateEntry(principal.getName("RFC2253"), certificate);
+            }
+            return keyStore;
+        }
+
+        private static KeyStore loadKeyStore(File certificateChainFile, File privateKeyFile,
+                Optional<String> keyPassword)
+                throws IOException, GeneralSecurityException {
+            PKCS8EncodedKeySpec encodedKeySpec = readPrivateKey(privateKeyFile, keyPassword);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'encodedKeySpec' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=3712629)